cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericColumnAggregator.java

```diff
@@ -183,7 +195,7 @@ public void aggregate(NGSIEvent event) {
                 if (lastData.containsKey(getLastDataUniqueKeyOnAggragation())) {
                     ArrayList<JsonElement> list = lastData.get(getLastDataUniqueKeyOnAggragation());
                     for (int i = 0 ; i < list.size() ; i++) {
-                        if (list.get(i).getAsString().equals(currentEntityId)) {
+                        if (list.get(i) !=null && list.get(i).getAsString().equals(currentEntityId)) {
                             long storedTS = CommonUtils.getTimeInstantFromString(
                                     lastData.get(getLastDataTimestampKeyOnAggregation()).
                                     get(i).getAsString());
```

Comment:
Not sure if the guards for not null in this file are actually needed... While I was working on unit tests I got some null pointer exceptions but I'm not sure if they can raise in real usage conditions.

Anyway, I think it doesn't hurt, as extra security measure in the code.

Reply:
NTC (informative)